packages/l10n/src/create-l10n-injector/create-l10n-injector.tsx

```diff
@@ -11,8 +11,76 @@ type InjectorOptions<LoadedData> = {
     cb: (error?: Error, data?: LoadedData) => void
   ) => void;
 };
+type State<Data> = {
+  isLoading: boolean;
+  data?: Data;
+  error?: Error;
+};
+type Action<Data> =
+  | { type: 'loading' }
+  | { type: 'ok'; data: Data }
+  | { type: 'error'; error: Error };
+
+const initialState = {
+  isLoading: true,
+};
+
+function reducer<Data>(
+  state = initialState,
+  action: Action<Data>
+): State<Data> {
+  switch (action.type) {
+    case 'loading':
+      return { isLoading: true };
+    case 'ok':
+      return { isLoading: false, data: action.data };
+    case 'error':
+      return { isLoading: false, error: action.error };
+    default:
+      return state;
+  }
+}
+
+export function createL10NHook<LoadedData>(
+  loadLocale: (
+    locale: string,
+    cb: (error?: Error, data?: LoadedData) => void
+  ) => void
+) {
+  return (locale: string) => {
+    const [data, dispatch] = React.useReducer<
+      (
+        prevState: State<LoadedData>,
+        action: Action<LoadedData>
+      ) => State<LoadedData>
+    >(reducer, initialState);
+
+    React.useEffect(() => {
+      let cleaning = false;
+      dispatch({ type: 'loading' });
```

Comment:
Nit: define `const setLoading = () => dispatch({...})` and `const setError =...` ontop.

Reply:
Any particular reason for doing it like this?

Reply:
Readability. It's a nit.

Reply:
I see. I think I prefer to keep it like this, the hook function is already a bit long